secrets/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// return codes:
+// 1: failed to read secret from env
+// 2: failed to parse hierarchical secret
+// 3: failed to write secret content into file
+func main() {
+	for _, name := range os.Args[1:] {
+		i := strings.Index(name, ":")
+		var keys []string
+		if i > 0 {
+			keys = strings.Split(name[i+1:], ",")
+			name = name[:i]
```

Comment:
Where does this `:` and comma-separated format come from? Is that something we defined, or something coming from  ECS?

Wondering, be cause we should try to avoid defining new micro-formats if possible (they've been ambiguous in the past)

Reply:
internal use only, to configure init container with secrets to be bound

Reply:
I updated this PR so that args are passed to the init container as a json-marshalled struct